### ISSUE TITLE
OCP 4.4 fails to bootstrap (SPIKE)

### ISSUE THREAD

**Issue by StevenBarre** [BCDevOps/OpenShift4-RollOut#271]
Testing our install scripts on OCP 4.4 fails to bootstrap. Opened a case with Red Hat https://access.redhat.com/support/cases/#/case/02689446

**Comment by StevenBarre** [comment]
Problem 1: ETCD will bootstrap itself using the first NIC, even if it's not the "production" interface. Tracked in [rhbz#1846093](https://bugzilla.redhat.com/show_bug.cgi?id=1846093) to possibly be fixed in 4.6.

Workaround for VMs is to just order the NICs correctly in the config options

```
ethernet0.pciSlotNumber = 192
ethernet1.pciSlotNumber = 224
ethernet2.pciSlotNumber = 256
scsi0.pciSlotNumber = 160
scsi1.pciSlotNumber = 1184
```

Workaround for Physicals is to disable DHCP for non-prod interfaces until the ETCD bootstrap is complete. MGMT and STOR networks are not needed that early in the cluster build.

Problem 2: NetworkManager race condition causing host to be named `localhost`. NM declaires itself "online" before completing getting its hostname from DHCP. Kubelet then starts and uses `localhost` as the hostname causing ETCD to become confused and have TLS cert errors. Tracked in [rhbz#1845885](https://bugzilla.redhat.com/show_bug.cgi?id=1845885) to possibly be fixed in 4.6.

Workaround: The [MCO PR](https://github.com/openshift/machine-config-operator/pull/1813/files)  purposed to fix this simply creates another systemd unit to delay `network-online.target` until the hostname is not `localhost`. As I already have code to adjust the Ignition files used to build the servers I can inject that systemd unit myself during bootstrap.

Next step: Test in KLAB.

**Comment by StevenBarre** [comment]
Fix tested in KLAB. Code updated.
